### PR TITLE
Add Svelte layer

### PR DESCRIPTION
**Pull request by msuess**
This PR adds support for Svelte based on the excellent Vue layer by Thanh Vuong.

### CODE REVIEW

**Comment by msuess** [comment]
Thanks for the review! I'm assuming you mean the double newlines after the first comment? I can remove them if this is blocking the merge, but I just copied most of the code over from the vue layer, where they are also present.

File: layers/+frameworks/svelte/config.el
Comment (smile13241324) [inline layers/+frameworks/svelte/config.el:1]: This should be svelte layer not vue
```suggestion
;;; config.el --- svelte layer config file for Spacemacs. -*- lexical-binding: t -*-
```

File: layers/+frameworks/svelte/config.el
Comment (smile13241324) [inline layers/+frameworks/svelte/config.el:5]: You should be the author :)

File: layers/+frameworks/svelte/funcs.el
Comment (smile13241324) [inline layers/+frameworks/svelte/funcs.el:5]: You should be the author ;)

File: layers/+frameworks/svelte/funcs.el
Comment (smile13241324) [inline layers/+frameworks/svelte/funcs.el:71]: This should rather be set in the users dotfile

File: layers/+frameworks/svelte/layers.el
Comment (smile13241324) [inline layers/+frameworks/svelte/layers.el:1]: Should be svelte layer

File: layers/+frameworks/svelte/layers.el
Comment (smile13241324) [inline layers/+frameworks/svelte/layers.el:5]: You should be the author

File: layers/+frameworks/svelte/layers.el
Comment (smile13241324) [inline layers/+frameworks/svelte/layers.el:24]: You should declare a dependency on lsp layer when the backend lsp is used

**Review by smile13241324 — CHANGES_REQUESTED** [review]
I have reviewed your commit and found some points you need to double check before this can be merged.

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop as I was doing some refactoring anyway I have fixed the open points on the go.

**Comment by msuess** [comment]
Thank you!

**Pull request closed without merging**
